clientv3/client.go

```diff
@@ -32,16 +32,25 @@ var (
 	ErrNoAvailableEndpoints = errors.New("etcdclient: no available endpoints")
 )
 
-// Client provides and manages an etcd v3 client session.
-type Client struct {
+type grpcAPI struct {
+	// Cluster is the cluster API for the client's connection.
+	Cluster pb.ClusterClient
 	// KV is the keyvalue API for the client's connection.
 	KV pb.KVClient
 	// Lease is the lease API for the client's connection.
 	Lease pb.LeaseClient
 	// Watch is the watch API for the client's connection.
 	Watch pb.WatchClient
-	// Cluster is the cluster API for the client's connection.
-	Cluster pb.ClusterClient
+}
+
+// Client provides and manages an etcd v3 client session.
+type Client struct {
+	Cluster
+	KV
+	Lease
+	Watcher
+
+	Grpc grpcAPI
```

Comment:
> should the test infra build the grpc stuff from the client's active connection

This is the correct way in my opinion. The server test should not really rely on client logic.